src/client/Microsoft.Identity.Client/Platforms/Android/Broker/AndroidBrokerHelper.cs

```diff
@@ -318,7 +318,28 @@ public async Task InitiateBrokerHandshakeAsync(Activity callerActivity)
                             return;
                         }
 
-                        throw new MsalClientException("Could not negotiate protocol version with broker.");
+                        dynamic errorResult = JObject.Parse(bundleResult?.GetString(BrokerConstants.BrokerResultV2));
+                        string errorCode = null;
+                        string errorDescription = null;
+
+                        if (!string.IsNullOrEmpty(errorResult))
+                        {
+                            errorCode = errorResult[BrokerResponseConst.BrokerErrorCode]?.ToString();
+                            errorDescription = errorResult[BrokerResponseConst.BrokerErrorMessage]?.ToString();
+
+                            _logger.Error($"An error occurred during hand shake with the broker. Error: {errorCode} Error Description: {errorDescription}");
+
+                        }
+                        else
+                        {
+                            errorCode = BrokerConstants.BrokerUnknownErrorCode;
+                            errorDescription = "An error occurred during hand shake with the broker, no detailed error information was returned";
+
+                            _logger.Error(errorDescription);
+                        }
+
+                        throw new MsalClientException(errorCode, 
+                            "Could not negotiate protocol version with broker due to the following error: " + errorDescription);
```

Comment:
>"Could not negotiate protocol version with broker due to the following error: " [](start = 28, length = 79)

is this needed? Why not "just" the error description you already collected above?

Reply:
you are right. that info is not needed.